src/Type/Php/HashHmacFunctionsReturnTypeExtension.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\UnionType;
+
+final class HashHmacFunctionsReturnTypeExtension implements DynamicFunctionReturnTypeExtension
+{
+
+	/** @var string[] */
+	private array $functions = [
+		'hash_hmac',
+		'hash_hmac_file',
+	];
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return in_array($functionReflection->getName(), $this->functions, true);
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		$algo = $functionCall->args[0]->value;
+		if (!in_array($algo, hash_hmac_algos(), true)) {
```

Comment:
There needs to be a fallback for PHP 7.1 using https://www.php.net/manual/en/function.hash-algos.php

Reply:
I have added `function_exists('hash_hmac_algos') ? hash_hmac_algos() : hash_algos()` check but sadly PHPStan doesn't care. Any suggestion?

Reply:
It's fine, just make the tests pass and I'll fix the PHPStan issue just before merging :) Thanks.